src/Core/src/Platform/iOS/KeyboardAutoManager.cs

```diff
@@ -0,0 +1,162 @@
+﻿/*
+ * This class is adapted from IQKeyboardManager which is an open-source
+ * library implemented for iOS to handle Keyboard interactions with
+ * UITextFields/UITextViews. Link to their MIT License can be found here:
+ * https://github.com/hackiftekhar/IQKeyboardManager/blob/7399efb730eea084571b45a1a9b36a3a3c54c44f/LICENSE.md
+ */
+
+using System.Collections.Generic;
+using UIKit;
+
+namespace Microsoft.Maui.Platform;
+
+internal static class KeyboardAutoManager
+{
+	// you can provide a topView argument or it will try to find the ContainerViewController
+	internal static void GoToNextResponderOrResign(UIView view, UIView? topView = null)
+	{
+		if (!view.CheckIfEligible())
+		{
+			view.ResignFirstResponder();
+			return;
+		}
+
+		var targetSuperview = topView ?? view.GetViewController()?.GetContainerViewController();
+		if (targetSuperview is null)
+		{
+			view.ResignFirstResponder();
+			return;
+		}
+
+		var textFields = GetDeepResponderViews(targetSuperview);
+
+		view.MoveToNextField(textFields);
+	}
+
+	static bool CheckIfEligible(this UIView view)
+	{
+		if (view is UITextField field && field.ReturnKeyType == UIReturnKeyType.Next)
+			return true;
+		else if (view is UITextView)
+			return true;
+
+		return false;
+	}
+
+	static UIView? GetContainerViewController(this UIViewController controller)
```

Comment:
Can you move this one as well to https://github.com/dotnet/maui/blob/main/src/Core/src/Platform/iOS/ViewExtensions.cs ? And then make it take a generic for the type it's searching for?